libdeno/internal.h

```diff
@@ -98,7 +98,7 @@ class DenoIsolate {
   v8::Persistent<v8::Function> recv_;
   v8::StartupData snapshot_;
   v8::Persistent<v8::ArrayBuffer> global_import_buf_;
-  v8::Persistent<v8::ArrayBuffer> shared_ab_;
+  v8::Persistent<v8::SharedArrayBuffer> shared_ab_;
```

Comment:
Is there a test we could add that demonstrates it failing using ArrayBuffer? It's not clear to me why this is necessary.

Reply:
Nothing is failing, but it's not possible to use Atomics on a regular ArrayBuffer.
If you don't believe me, try this:

```
D:\deno3>node
> let ab = new ArrayBuffer(4);
undefined
> let u8 = new Uint8Array(ab)
undefined
> Atomics.add(u8, 0, 1)
TypeError: [object Uint8Array] is not an integer shared typed array.
    at Atomics.add (<anonymous>)
>
```

Reply:
You can modify this test, I think 
```c
TEST(LibDenoTest, Shared) {
  uint8_t s[] = {0, 1, 2};
  deno_buf shared = {nullptr, 0, s, 3};
  Deno* d = deno_new(deno_config{0, snapshot, shared, nullptr});
  EXPECT_TRUE(deno_execute(d, nullptr, "a.js", "Shared()"));
  deno_execute(d, nullptr, "a.js", "Shared()");
  EXPECT_EQ(nullptr, deno_last_exception(d));
  EXPECT_EQ(s[0], 42);
  EXPECT_EQ(s[1], 43);
  EXPECT_EQ(s[2], 44);
  deno_delete(d);
}
```

Reply:
Improved (but not working):

```c++
TEST(LibDenoTest, SharedAtomics) {
  int32_t s[] = {0, 1, 2};
  deno_buf shared = {nullptr, 0, reinterpret_cast<uint8_t*>(s), sizeof s};
  Deno* d = deno_new(deno_config{0, empty, shared, nullptr});
  deno_execute(d, nullptr, "a.js",
               "Atomics.add(new Int32Array(libdeno.shared), 0, 1)");
  EXPECT_EQ(nullptr, deno_last_exception(d));
  EXPECT_EQ(s[0], 1);
  EXPECT_EQ(s[1], 1);
  EXPECT_EQ(s[2], 2);
  deno_delete(d);
}
```